packages/amplify-cli-core/API.md

```diff
@@ -747,6 +747,32 @@ export type EventPrefix = 'pre' | 'post';
 // @public (undocumented)
 export const executeHooks: (hooksMetadata: HooksMeta) => Promise<void>;
 
+// @public (undocumented)
+export type ExeInfo = {
+    amplifyMeta?: $TSAny;
+    awsConfigInfo?: $TSAny;
+    backendConfig?: $TSAny;
+    existingLocalAwsInfo?: $TSAny;
+    existingLocalEnvInfo?: $TSAny;
+    existingProjectConfig?: $TSAny;
+    existingTeamProviderInfo?: $TSAny;
+    forcePush?: boolean;
+    inputParams?: $TSAny;
+    isNewEnv?: boolean;
+    isNewProject?: boolean;
+    iterativeRollback?: boolean;
+    localEnvInfo?: $TSAny;
+    metaData?: $TSAny;
+    pinpointApp?: $TSAny;
+    pinpointClient?: $TSAny;
+    pinpointInputParams?: $TSAny;
+    projectConfig?: $TSAny;
+    restoreBackend?: boolean;
+    serviceMeta?: $TSAny;
+    sourceEnvName?: string;
+    teamProviderInfo?: $TSAny;
+};
```

Comment:
This type looks like set of randomly associated properties. Which might also be redundant with other data that's already in context.
Should we attempt to split it ? Or at least create types for property groupings that make some sense together ?

Reply:
This work is primarily here to facilitate the code coverage work Spencer and I are working on. Refactoring it better should absolutely be a future task. Right now I'm writing things almost solely to understand and be able to mock what's in exeInfo.

Reply:
Yes, but this type is going to spread around pretty fast, so we should get this right.
I suggest we spend some time understanding consumption patterns of this and if they're orthogonal we should type it as union of smaller types.

Reply:
Updated types for better bucketing derived from explicitly logical and derived from an analysis of instantiation across all files. I don't think this is perfect, but I think it's better than what I had.